universite-du-quebec-a-montreal.csl

```diff
@@ -333,15 +521,41 @@
       </else-if>
       <else-if type="speech" match="any">
         <group delimiter=", ">
-          <text variable="genre"/>
+          <text variable="abstract"/>
+          <text variable="note"/>
+          <text variable="publisher-place"/>
+        </group>
+      </else-if>
+      <else-if type="manuscript" match="all" variable="issue">
+        <group delimiter=". ">
+          <text variable="issue" prefix="[" suffix="]"/>
+          <text variable="genre" prefix="[" suffix="]"/>
+          <group delimiter=" ">
+            <text variable="archive"/>
+            <text variable="archive_location" prefix="(" suffix=")"/>
+          </group>
           <text variable="publisher-place"/>
         </group>
       </else-if>
+      <else-if type="legal_case" match="any">
+        <choose>
+          <if match="any" variable="page">
+            <group delimiter=" ">
+              <text variable="authority"/>
+              <text variable="container-title"/>
+              <text variable="page"/>
+            </group>
+          </if>
+        </choose>
+      </else-if>
       <else>
-        <group delimiter=", ">
+        <group delimiter=". ">
           <choose>
-            <if variable="event" match="none" type="chapter patent post-weblog book article-journal post motion_picture broadcast">
+            <if variable="event" match="none" type="chapter patent post-weblog book article-journal post motion_picture broadcast map musical_score figure legislation dataset manuscript article webpage entry-encyclopedia">
               <text variable="genre" prefix="[" suffix="]"/>
+              <text variable="scale" prefix="Échelle : "/>
+              <text variable="number"/>
+              <text variable="annote"/>
```

Comment:
now you're using both `note` and `annote`? You're really not doing users a favor by making them use a style that requires data entry that breaks everything else.

Reply:
I will delete "annote" and make minimal use of "note".